src/Lecture1.hs

```diff
@@ -98,7 +105,12 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+
+subString :: Int -> Int -> String -> String
+subString start end str
+  | end < 0 = ""
+  | start < 0 && end >= 0 = take (end  + 1) str
```

Comment:
You don't need to check that `end` is greater or equal to zero because guards are checked from top to bottom so if the first guard failed, you already know that `end` is `>= 0` here 🙂 

Additionally, you can change check from `start < 0` to `start <= 0` to avoid calling `drop` first on another corner case 👍🏻 

```suggestion
  | start <= 0 = take (end  + 1) str
```